### PR TITLE
Add chefspec testing for sc-mongodb::default

### PR DESCRIPTION
**Pull request by shortdudey123**

### CODE REVIEW

**Comment by josephholsten** [comment]
Dense. It'll take me a bit to review.

File: recipes/install.rb [hunk 1]
```diff
@@ -31,11 +31,12 @@
 config_type = node['mongodb']['is_mongos'] ? 'mongos' : 'mongod'
 
 # prevent-install defaults, but don't overwrite
-file node['mongodb']['sysconfig_file'][config_type] do
+file "#{node['mongodb']['sysconfig_file'][config_type]} install" do
```
Comment (josephholsten) [inline recipes/install.rb]: Shall we extract a variable `sysconfig_file` so these lines read more easily?

File: recipes/install.rb [hunk 2]
```diff
@@ -55,7 +56,7 @@
 end
 
 # and we install our own init file
-if node['mongodb']['apt_repo'] == 'ubuntu-upstart'
+if node['platform'] == 'ubuntu' && node['platform_version'].to_f < 15.04
```
Comment (josephholsten) [inline recipes/install.rb]: I thought there was a dsl helper for this check, but it's still fine this way.

File: recipes/install.rb [hunk 3]
```diff
@@ -82,8 +83,7 @@
     sysconfig_file: node['mongodb']['sysconfig_file'][config_type],
     ulimit: node['mongodb']['ulimit'],
     bind_ip: node['mongodb']['config'][config_type]['net']['bindIp'],
-    port: node['mongodb']['config'][config_type]['net']['port'],
-    user: node['mongodb']['user']
+    port: node['mongodb']['config'][config_type]['net']['port']
```
Comment (josephholsten) [inline recipes/install.rb]: Seems like there are a lot of things pulling from `node.mongodb.config[config_type]`, should we extract variable?

File: spec/recipes/default_spec.rb [hunk 4]
```diff
@@ -0,0 +1,384 @@
+require 'spec_helper'
+
+describe 'sc-mongodb::default' do
+  context 'with default node attributes' do
+    let(:mongod_conf_rhel) do
```
Comment (josephholsten) [inline spec/recipes/default_spec.rb]: Maybe a comment explaining why these aren't inline a in the individual tests? I try to keep unit tests as individually self contained as possible, so this means I've got to read two places to grok the test. 

If we can just extract the parts of tests that use this to only be in one place, would that balance the fragility and ease-of-jumping-into-tests?

File: recipes/install.rb (hunk 2 above)
Comment (shortdudey123) [inline recipes/install.rb]: Unfortunately now :(

**Pull request merged**